File: Rust/easy/1502_can_make_arithmetic_progression_from_sequence.rs
//Given an array of numbers arr. A sequence of numbers is 
//called an arithmetic progression if the difference between
//any two consecutive elements is the same.
//Return true if the array can be rearranged to form an arithmetic
//progression, otherwise, return false.
impl Solution {
    pub fn can_make_arithmetic_progression(arr: Vec<i32>) -> bool {
        let mut arr_copy = arr.clone();
        arr_copy.sort();
        let mut dx: i32 = arr_copy[1] - arr_copy[0];
        for idx in 2..arr_copy.len() {
            if arr_copy[idx] - arr_copy[idx - 1] != dx {
                return false
            }
        }
        true
    }
}

File: Python/easy/1502_can_make_arithmetic_progression_from_sequence.py
"""
Given an array of numbers arr. A sequence of numbers is called an arithmetic progression
if the difference between any two consecutive elements is the same.
Return true if the array can be rearranged to form an arithmetic progression, otherwise,
return false.
"""
from typing import List


class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr = sorted(arr)
        dx = arr[1] - arr[0]
        for idx in range(2, len(arr)):
            if arr[idx] - arr[idx - 1] != dx:
                return False
        return True


if __name__ == "__main__":
    arr = [3, 5, 1]
    out = True
    res = Solution().canMakeArithmeticProgression(arr)
    assert res == out, (res, out)

    arr = [1, 2, 4]
    out = False
    res = Solution().canMakeArithmeticProgression(arr)
    assert res == out, (res, out)
